src/test/java/org/sonatype/gradle/plugins/scan/nexus/iq/index/NexusIqIndexTaskTest.java

```diff
@@ -0,0 +1,106 @@
+
+import org.sonatype.gradle.plugins.scan.common.DependenciesFinder;
+
+import org.gradle.api.Project;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anySet;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.sonatype.gradle.plugins.scan.nexus.iq.index.NexusIqIndexTask.MODULE_XML_FILE;
+import static org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqPluginScanExtension.SONATYPE_CLM_FOLDER;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NexusIqIndexTaskTest
+{
+  @Mock
+  private DependenciesFinder dependenciesFinderMock;
+
+  @Mock
+  private ModuleIoManager moduleIoManagerMock;
+
+  @Test
+  public void testSaveModule_singleModule() throws IOException {
+    Module module = new Module()
+        .setId("test-module")
+        .setPathname("test-module");
+    File file = Paths.get("test-module", "build", SONATYPE_CLM_FOLDER, MODULE_XML_FILE).toFile();
+
+    when(dependenciesFinderMock.findModules(any(Project.class), eq(false), anySet()))
+        .thenReturn(Collections.singletonList(module));
+    doNothing().when(moduleIoManagerMock).writeModule(file, module);
+
+    NexusIqIndexTask task = buildIndexTask();
+    task.setDependenciesFinder(dependenciesFinderMock);
+    task.setModuleIoManager(moduleIoManagerMock);
+    task.saveModule();
+
+    verify(dependenciesFinderMock).findModules(any(Project.class), eq(false), anySet());
+    verify(moduleIoManagerMock).writeModule(file, module);
+  }
+
+  @Test
+  public void testSaveModule_multipleModules() throws IOException {
+    Module module1 = new Module()
+        .setId("test-module-1")
+        .setPathname("test-module-1");
+    Module module2 = new Module()
+        .setId("test-module-2")
+        .setPathname("test-module-2");
+    File file1 = Paths.get("test-module-1", "build", SONATYPE_CLM_FOLDER, MODULE_XML_FILE).toFile();
+    File file2 = Paths.get("test-module-2", "build", SONATYPE_CLM_FOLDER, MODULE_XML_FILE).toFile();
+
+    when(dependenciesFinderMock.findModules(any(Project.class), eq(false), anySet()))
+        .thenReturn(Arrays.asList(module1, module2));
+    doNothing().when(moduleIoManagerMock).writeModule(file1, module1);
+    doNothing().when(moduleIoManagerMock).writeModule(file2, module2);
+
+    NexusIqIndexTask task = buildIndexTask();
+    task.setDependenciesFinder(dependenciesFinderMock);
+    task.setModuleIoManager(moduleIoManagerMock);
+    task.saveModule();
+
+    verify(dependenciesFinderMock).findModules(any(Project.class), eq(false), anySet());
+    verify(moduleIoManagerMock).writeModule(file1, module1);
+    verify(moduleIoManagerMock).writeModule(file2, module2);
+  }
```

Comment:
Could it be worth adding a test which has some modules excluded as well? I mean I see `DependenciesFinder.findModules` which uses the excludes already existed, but it could be more to test that your configuration `NexusIqPluginIndexExtension` accepts them and passes them to that method in `NexusIqIndexTask.saveModule`

Reply:
> Could it be worth adding a test which has some modules excluded as well? I mean I see `DependenciesFinder.findModules` which uses the excludes already existed, but it could be more to test that your configuration `NexusIqPluginIndexExtension` accepts them and passes them to that method in `NexusIqIndexTask.saveModule`

Done: 77ba696b8fe11a816e4bed62bdcd1b75495e821b, 34c00d70765afbb3da3aa00f1cd6c6799933c2f1